ArraysAmbiguousSNPMiscallingMar2019/filter_vcf.sh

```diff
@@ -0,0 +1,135 @@
+#  invocation:                                                                                                        #
+#  filter_vcf.sh <GATK> <INTERVAL_LIST> <VCF>                                                                         #
+#                                                                                                                     #
+#  where:                                                                                                             #
+#     <GATK> points to your gatk executable                                                                           #
+#     <INTERVAL_LIST> points to one of the interval lists provided in the IntervalLists directory                     #
+#     <VCF> is a vcf that needs it's bottom-stranded ambiguous SNPs filtered out (they will )                         #
+#           in the VCF, but their filter field will contain the string ARRAY_AMBIGUOUS_SNP_BUG                        #
+#                                                                                                                     #
+# For example (from within the ArraysAmbiguousSNPMiscallingMar2019 directory):                                        # 
+# ./filter_vcf.sh ~/gatk/gatk IntervalLists/PsychChip_v1-1_15073391_A1.1.3.interval_list Psych_cohort.vcf.gz          #
+#                                                                                                                     #
+#  After filtering, the script runs a sanity check to see that the correct number of sites                            #
+#  have been filtered. If the script ran successfully, A message in the end should tell you so.                       # 
+#                                                                                                                     #
+#  The script attempts to clean up after itself but leaves log files around.                                          #
+#                                                                                                                     #
+#######################################################################################################################
+
+set -eo pipefail
+
+GATK=$1
+INTERVAL_LIST=$2
+VCF=$3
+
+FILTER=ARRAY_AMBIGUOUS_SNP_BUG
+DICT=$(basename ${INTERVAL_LIST%interval_list}dict)
+BED=$(basename ${INTERVAL_LIST%interval_list}bed)
+VCF_OUT=$(basename ${VCF%vcf*}filtered.vcf${VCF##*vcf})
+VCF_OUT_SNPS=$(basename ${VCF%vcf*}filtered.SNPs.vcf${VCF##*vcf})
+VCF_SNPS=$(basename ${VCF%vcf*}SNPs.vcf${VCF##*vcf})
+VCF_NONSNPS=$(basename ${VCF%vcf*}NONSNPs.vcf${VCF##*vcf})
+VCF_UNMERGED_IL=vcf_unmerged.interval_list
+VCF_MERGED_IL=vcf_merged.interval_list
+FILTERED_VARIANTS=filtered.variants.vcf.gz
+
+echo "indexing vcf"
+$GATK IndexFeatureFile -F $VCF 2> index.vcf.log || (cat index.vcf.log && exit 1)
+
+echo "grabbing dictionary from interval list"
```

Comment:
Maybe there's a tool for this?  Didn't realize it was as simple as that.

Reply:
there's no tool since this is a bad usecase...we should be able to use interval_lists as masks in GATK...but they need a tribble codec https://github.com/broadinstitute/gatk/issues/5788

Reply:
once we go to bed, we are not sure that the dictionaries match...so I had to extract it by hand...

Reply:
actually...it doesn't do what I want...:-(

Reply:
oh well.